### PR TITLE
Implementing grid

### PR DESCRIPTION
**Pull request by rbvea**
_requires db update_

## New functionality

- [x] User should be able to reorder items within a section
- [x] User should be able to dynamically specify width

## Bug fixes

- [x] User should be able to upload photo right after item creation
- [x] Photo data should save as valid json (not `[Object object]`).

## Testing
- [x] Test for bump_up function passes

Related tickets [#113595399 #114220315 #113889971 #113672853]

### CODE REVIEW

**Comment by rbvea** [comment]
@tommyokeefe can you check this out in the morning?

**Comment by tommyokeefe** [comment]
@rbvea for sure!

File: src/controllers/BrandCenter/BrandItemController.php [hunk 1]
```diff
@@ -64,6 +67,12 @@ public function edit($id, $post) {
       if($post['data']) {
         $item->setData($post['data']);
       }
+      if($post['width']) {
+        $item->setWidth($post['width']);
+      }
+      if(isset($post['listOrder'])) {
```
Comment (rbvea) [inline src/controllers/BrandCenter/BrandItemController.php]: listOrder was 0 which is false in php...

File: style.css [hunk 2]
```diff
@@ -2728,61 +2784,61 @@ html[_v-853629ac] {
...
-.item--photo:nth-child(1n) {
+.item--full:nth-child(1n) {
   margin-right: 30px;
 }
-.item--photo:last-child {
+.item--full:last-child {
   margin-right: 0;
 }
-.item--photo:nth-child(2n) {
+.item--full:nth-child(undefinedn) {
```
Comment (rbvea) [inline style.css]: Hmm, undefined?

**Comment by rbvea** [comment]
Finally got the build to pass.  Looks like travis-ci only has phantomjs@1.x.x, also had a stray cssnano ref.

**Comment by rbvea** [comment]
@tommyokeefe can you check this out?

**Comment by tommyokeefe** [comment]
Crap. Totally forgot about this. I'll get on it now.

**Comment by tommyokeefe** [comment]
Looks like after adding/changing a photo you need a refresh for it to show. Not a big deal, but something that should be fairly easy to fix.

**Comment by rbvea** [comment]
Does clicking the 'save' buttow show the recently uploaded image?  I didn't toggle it back because if the user say, changed the width too, it doesn't save it.

**Comment by tommyokeefe** [comment]
No, after clicking save it shows up as blank until I refresh:
saveimage

**Comment by rbvea** [comment]
Alright, will check out that out, thanks!

**Comment by tommyokeefe** [comment]
No problem!

**Comment by tommyokeefe** [comment]
Otherwise this looks good. I love having the travis.ci stuff here. Pretty sweet!

:+1:

**Comment by rbvea** [comment]
I forgot to ask, is that on the production instance?   Could it be caching?

**Comment by tommyokeefe** [comment]
I tested it locally. I can look on prod too.

**Comment by rbvea** [comment]
I've tested it on my local and it looks that way too, will check it out (seems like it's a problem with the save function overriding the photo data [if you just toggle the edit button after uploading you should see it persisted])

**Comment by tommyokeefe** [comment]
Yeah, same thing is happening on prod. If you look at the html, the src tag for the image is blank:

```
<section _v-853629ac="" _v-510c3030="" class="item item--photo item--half" id="item-62">
  <div class="item-wrapper" _v-853629ac="">
    <a href="" _v-853629ac="" class="edit mdl-button mdl-button--raised">Edit</a>
    <!--v-if-->
    <!--v-if-->
    <div _v-853629ac="">
      <!--v-if-->
      <img _v-853629ac="" src="">
      <!--v-if-->
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</section>
```

**Comment by rbvea** [comment]
@tommyokeefe that should do it, I have a bad habit of using `delete` and mutating the object...

**Comment by rbvea** [comment]
Updated production: not seeing the bug (cache issues not withstanding)

http://dev-wordpress-boilerplate.pantheon.io/brand-center/#!/page

**Comment by tommyokeefe** [comment]
You're right. I got it working now. Looks good! :+1:

**Pull request merged**